src/Microsoft.SqlTools.ServiceLayer/EditData/UpdateManagement/RowCreate.cs

```diff
@@ -88,13 +89,72 @@ public override DbCommand GetCommand(DbConnection connection)
         {
             Validate.IsNotNull(nameof(connection), connection);
 
-            // Build the script and generate a command
-            ScriptBuildResult result = BuildInsertScript(forCommand: true);
+            // Process the cells and columns
+            List<string> declareColumns = new List<string>();
+            List<string> inColumnNames = new List<string>();
+            List<string> outClauseColumnNames = new List<string>();
+            List<string> inValues = new List<string>();
+            List<SqlParameter> inParameters = new List<SqlParameter>();
+            List<string> selectColumns = new List<string>();
+            for(int i = 0; i < AssociatedObjectMetadata.Columns.Length; i++)
+            {
+                DbColumnWrapper column = AssociatedResultSet.Columns[i];
+                EditColumnMetadata metadata = AssociatedObjectMetadata.Columns[i];
+                CellUpdate cell = newCells[i];
+                
+                // Add the output columns regardless of whether the column is read only
+                outClauseColumnNames.Add($"inserted.{metadata.EscapedName}");
+                declareColumns.Add($"{metadata.EscapedName} {ToSqlScript.FormatColumnType(column, useSemanticEquivalent: true)}");
+                selectColumns.Add(metadata.EscapedName);
+
+                // Continue if we're not inserting a value for this column
+                if (!IsCellValueProvided(column, cell, DefaultValues[i]))
+                {
+                    continue;
+                }
+                
+                // Add the input column
+                inColumnNames.Add(metadata.EscapedName);
+                
+                // Add the input values as parameters
+                string paramName = $"@Value{RowId}_{i}";
+                inValues.Add(paramName);
+                inParameters.Add(new SqlParameter(paramName, column.SqlDbType) {Value = cell.Value});
+            }
+            
+            // Put everything together into a single query            
+            // Step 1) Build a temp table for inserting output values into
+            string tempTableName = $"@Insert{RowId}Output";
+            string declareStatement = string.Format(DeclareStatement, tempTableName, string.Join(", ", declareColumns));
+
+            // Step 2) Build the insert statement
+            string joinedOutClauseNames = string.Join(", ", outClauseColumnNames);
+            string insertStatement = inValues.Count > 0
+                ? string.Format(InsertOutputValuesStatement, 
+                    AssociatedObjectMetadata.EscapedMultipartName,
+                    string.Join(", ", inColumnNames), 
+                    joinedOutClauseNames,
+                    tempTableName,
```

Comment:
>tempTableName [](start = 20, length = 13)

did you happen to investigate how SSMS supports this scenario?  I wonder if using a temp table is the simplest way to fix this bug?

Reply:
SSMS doesn't do update/insert using an OUTPUT clause. As such it sidesteps the issue entirely, but raises the issue of how to handle calculated column values. Using a temp table is the simplest way to address the issue of OUTPUT clauses on tables with triggers according to various posts on the internet.